### DISCUSSION TITLE
:pray:  Add yaml format to import features

### DISCUSSION THREAD

**Discussion by adriens** [duckdb/duckdb#6872] [Ideas]
# :grey_question:  About

More and more data are avilable as `yaml` content, it would be very convenient to add  its support in [Importing Data](https://duckdb.org/docs/data/overview)

:bulb: We could use it oser `HTTPS` like for `json`

# :heavy_check_mark:  Current supported formats

- [x] `csv`
- [x] `json`
- [ ] `yaml`
- [ ] `xml`

**Comment by Mause** [comment]
Can you give an example of a dataset that is only available via yaml? To my knowledge it's mostly used as a config format, rather than as a data interchange format

**Reply by adriens** [reply]
Sure, in fact, if you consider that manifests are data, here are some : https://github.com/teaxyz/pantry/blob/main/projects/ansible.com/package.yml

Implementin yaml would it make possible to make datascience on configurations, and in K8S context, that seems realyy useful (my opitnion) :smile_cat:

**Comment by teaguesterling** [comment]
Apologies in advance for digging this discussion out of the depths of 2 years ago.

I also have a need to process large numbers of YAML files via DuckDB, and have been resorting to translating them to JSON first. I recently wanted to see if I could step it up a bit and decided to create a `duckdb_yaml` extension, that is currently in a "basic functionality working" state. On the off chance it's still useful to you @adriens: https://github.com/teaguesterling/duckdb_yaml/

As a side note: this is probably of somewhat below-average code quality. As noted in the README, this as as much an exercise in seeing if I could develop a DuckDB extension entirely (or near entirely) via an LLM. It just finished passing its own tests.
